File: src/main/kotlin/dto/file/FolderInfo.kt
package org.zak.dto.file

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.*

data class FolderInfo(
	val name: String,
	val fullPath: String,
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
	val lastModified: Date,
	val size: Long,
	val readableSize: String,
	val itemCount: Int,
	val isDirectory: Boolean = true
)

File: angular_server/src/app/core/model/folder-info.ts
export class FolderInfo{
	constructor(
		public name: string,
		public fullPath: string,
		public lastModified: Date,
		public size: bigint,
		public extension: string,
		public readableSize: string,
		public itemCount: bigint,
		public isDirectory: boolean = true
	){}
}